packages/frontend-core/src/components/grid/cells/HeaderCell.svelte

```diff
@@ -118,6 +120,33 @@
     open = false
   }
 
+  const duplicateColumn = async () => {
+    open = false
+
+    // Generate new name
+    let newName = `${column.name} copy`
+    let attempts = 2
+    while ($definition.schema[newName]) {
+      newName = `${column.name} copy ${attempts++}`
+    }
+
+    // Save schema with new column
+    const existingColumnDefinition = $definition.schema[column.name]
+    await datasource.actions.saveDefinition({
+      ...$definition,
+      schema: {
+        ...$definition.schema,
+        [newName]: {
+          ...existingColumnDefinition,
+          name: newName,
+          schema: {
+            ...existingColumnDefinition.schema,
```

Comment:
If we're already spreading the `existingColumnDefinition` on line 140, so we need specify the schema property here?

Reply:
We need to change the schema from the existing column definition, so this is how you copy and change a nested property without mutating the base! If I didn't spread the base schema then schema would only have whatever properties specified here.